extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -185,6 +237,28 @@ export class AmpIframe extends AMP.BaseElement {
     return loadPromise(iframe);
   }
 
+  /** @override  */
+  viewportCallback(inViewport) {
+    // Lets the iframe know that it became visible or no longer is.
+    this.sendIntersection_();
```

Comment:
Just make this entire method:

`this.intersectionObserver_.viewportCallback(inViewport)`

The logic can then go into that method and not be duplicated with amp-ad